hdw-a339x/src/systems/tcas/src/components/TcasComputer.ts

```diff
@@ -297,67 +296,80 @@ export class TcasComputer implements TcasComponent {
     this.rateToMaintain = new LocalSimVar('L:A32NX_TCAS_RA_RATE_TO_MAINTAIN', 'number');
     this.upAdvisoryStatus = new LocalSimVar('L:A32NX_TCAS_RA_UP_ADVISORY_STATUS', 'Enum');
     this.downAdvisoryStatus = new LocalSimVar('L:A32NX_TCAS_RA_DOWN_ADVISORY_STATUS', 'Enum');
-    this.selectedTrafficDataSource = ({"NONE": 0, "SIM": 1, "VATSIM": 2})[NXDataStore.get("CONFIG_TRAFFIC_SOURCE", "NONE")];
+    this.selectedTrafficDataSource = ({"NONE": 0, "SIM": 1, "VATSIM": 2, "IVAO": 3})[NXDataStore.get("CONFIG_TRAFFIC_SOURCE", "NONE")];
     SimVar.SetSimVarValue('L:A32NX_TRAFFIC_SELECTOR_SOURCE', 'number', this.selectedTrafficDataSource);
     SimVar.SetSimVarValue('L:A32NX_TRAFFIC_SELECTOR_DISPLAY', 'number', NXDataStore.get("CONFIG_TRAFFIC_DISPLAY_DEFAULT", "NO") === "YES" ? 1 : 0);
     this.airTraffic = [];
     this.raTraffic = [];
     this.sensitivity = new LocalSimVar('L:A32NX_TCAS_SENSITIVITY', 'number');
     this.sensitivity.setVar(1);
     this.updateThrottler = new UpdateThrottler(TCAS.REFRESH_RATE); // P5566074 pg 11:45
-    this.vatsimDataThrottler = null;
     this.inhibitions = Inhibit.NONE;
     this.ppos = { lat: NaN, long: NaN };
     this._newRa = new ResAdvisory(null, false, 0, false);
     this.advisoryState = TcasState.NONE;
     this.sendAirTraffic = [];
     this.activeRa = new ResAdvisory(null, false, 0, false);
     this.soundManager = new TcasSoundManager();
+    this.secondsSinceLastTrafficDataUpdate = 0;
   }
 
-  updateVatsimData() {
+  private updateTrafficData(_deltaTime: number): void {
     if (this.selectedTrafficDataSource === 0) {
-      this.vatsimDataThrottler = null;
       return;
     }
+
+    this.secondsSinceLastTrafficDataUpdate += _deltaTime / 1000;
+    if(this.secondsSinceLastTrafficDataUpdate < 5) {
+      return;
+    }
+
+    this.secondsSinceLastTrafficDataUpdate = 0;
+
     const port = SimVar.GetSimVarValue('L:A339X_TRAFFIC_PORT', 'number');
     if (port === 0 || this.sendAirTraffic.length === 0) {
-      this.vatsimDataThrottler = setTimeout(() => this.updateVatsimData(), 1000 * 5);
       return;
     }
-    const list = this.airTraffic.filter((entry) => entry.alive && entry.isDisplayed);
+    const list = this.airTraffic.filter((traffic) => traffic.alive === true && this.sendAirTraffic.some(x => x.ID === traffic.ID))
     if (list.length) {
       Promise.all(
         list.map(
           (traffic) =>
             new Promise((resolve) => {
-              const existing = traffic.vatsimEntry;
+              const existing = traffic.trafficData;
               fetchPilotInfo(
                 traffic.ID,
                 port,
-                traffic.vatsimFetchCount < 5 && typeof existing === 'object' && existing !== null,
+                traffic.dataFetchCount < 5 && typeof existing === 'object' && existing !== null,
```

Comment:
i need to remove this in the future, it was an attempt at reducing request time but didn't work

Reply:
@liz3 do you want me to take care on this with this changes?

Reply:
yes i will